flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java

```diff
@@ -465,27 +514,23 @@ private static void start(YarnConfiguration conf, String principal, String keyta
 
 			File flinkConfDirPath = findFile(flinkDistRootDir, new ContainsName(new String[]{"flink-conf.yaml"}));
 			Assert.assertNotNull(flinkConfDirPath);
+			org.apache.flink.configuration.Configuration flinkCfg =
+					new org.apache.flink.configuration.Configuration();
 
 			if (!StringUtils.isBlank(principal) && !StringUtils.isBlank(keytab)) {
+
 				//copy conf dir to test temporary workspace location
 				tempConfPathForSecureRun = tmp.newFolder("conf");
 
 				String confDirPath = flinkConfDirPath.getParentFile().getAbsolutePath();
 				FileUtils.copyDirectory(new File(confDirPath), tempConfPathForSecureRun);
 
-				try (FileWriter fw = new FileWriter(new File(tempConfPathForSecureRun, "flink-conf.yaml"), true);
-					BufferedWriter bw = new BufferedWriter(fw);
-					PrintWriter out = new PrintWriter(bw)) {
-
-					LOG.info("writing keytab: " + keytab + " and principal: " + principal + " to config file");
-					out.println("");
-					out.println("#Security Configurations Auto Populated ");
-					out.println(SecurityOptions.KERBEROS_LOGIN_KEYTAB.key() + ": " + keytab);
-					out.println(SecurityOptions.KERBEROS_LOGIN_PRINCIPAL.key() + ": " + principal);
-					out.println("");
-				} catch (IOException e) {
-					throw new RuntimeException("Exception occured while trying to append the security configurations.", e);
-				}
+				flinkCfg.setString(SecurityOptions.KERBEROS_LOGIN_KEYTAB.key(), keytab);
+				flinkCfg.setString(SecurityOptions.KERBEROS_LOGIN_PRINCIPAL.key(), principal);
+				flinkCfg.setString(CoreOptions.MODE.key(), OLD_MODE);
```

Comment:
I think we should assign the `Configuration` to `flinkConfiguration`, then we don't have to read it in from disk again.

Reply:
flinkConfiguration is instance variable and the method here is static.

Reply:
Then let's store `flinkCfg` as a static variable and use it to create a new one
`flinkConfiguration = new Configuration(flinkCfg)`

Reply:
I make flinkConfiguration static. It's confusing to keep 2 flinkcfg, one static and one instance var.